ui/src/components/FormModifications/FormModifications.ts

```diff
@@ -0,0 +1,124 @@
+import { Mode } from '../../constants/modes';
+import { AcceptableFormValueOrNullish } from '../../types/components/shareableTypes';
+import { BaseFormState, AnyEntity, EntitiesWithModifications } from '../BaseFormTypes';
+import { MarkdownMessageProps } from '../MarkdownMessage/MarkdownMessage';
+
+export const handleStateFieldModificationProp = (
+    key: 'display' | 'value' | 'disabled' | 'markdownMessage',
+    propValue: string | number | boolean | object,
+    fieldId: string,
+    state: BaseFormState
+) => {
+    const shallowStateCopy = { ...state };
+    if (shallowStateCopy.data) {
+        shallowStateCopy.data[fieldId][key] = propValue as AcceptableFormValueOrNullish &
+            boolean &
+            (MarkdownMessageProps | undefined);
+        return {
+            changesOccured: true,
+            data: shallowStateCopy,
+        };
+    }
+    return {
+        changesOccured: false,
+        data: shallowStateCopy,
+    };
+};
+
+export const handleEntityModificationProp = (
+    key: 'help' | 'label',
+    propValue: string,
+    fieldId: string,
+    state: BaseFormState
+) => {
+    const shallowStateCopy = { ...state };
+    if (shallowStateCopy?.data?.[fieldId]) {
+        shallowStateCopy.data[fieldId].modifiedEntitiesData = {
+            ...shallowStateCopy.data[fieldId].modifiedEntitiesData,
+            [key]: propValue,
+        };
+        return {
+            changesOccured: true,
+            data: shallowStateCopy,
+        };
+    }
+
+    return {
+        changesOccured: false,
+        data: shallowStateCopy,
+    };
+};
+
+export function getAllFieldsWithModifications(entities: AnyEntity[]): EntitiesWithModifications[] {
+    const entitiesWithModifications = entities.filter(
+        (e) =>
+            e.type !== 'helpLink' &&
+            e.type !== 'checkboxGroup' &&
+            e.type !== 'oauth' &&
+            e.type !== 'custom' &&
+            e?.modifyFieldsOnValue
+    );
+
+    return entitiesWithModifications as EntitiesWithModifications[];
+}
+export const getModifiedState = (
+    state: BaseFormState,
+    mode: Mode,
+    entitiesToModify: EntitiesWithModifications[]
+) => {
+    let stateShallowCopy = { ...state };
+    let shouldUpdateState = false;
+
+    entitiesToModify?.forEach((entity: EntitiesWithModifications) => {
+        let modification = entity?.modifyFieldsOnValue?.find(
+            (mod) =>
+                stateShallowCopy?.data?.[entity.field]?.value === mod.fieldValue &&
+                (!mod.mode || mod.mode === mode)
+        );
+
+        if (!modification) {
+            modification = entity?.modifyFieldsOnValue?.find(
+                (mod) =>
+                    mod.fieldValue === '[[any_other_value]]' && (!mod.mode || mod.mode === mode)
```

Comment:
I would extract `[[any_other_value]]` to at least to the top of the file with great explanatory name

Reply:
added `const VALUE_TO_TRIGGER_UPDATE_FOR_ANY_NOT_LISTED_VALUES = '[[any_other_value]]';
`